components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/dto/TenantManagementEventDTO.java

```diff
@@ -0,0 +1,293 @@
+    public long getIat() {
+        return iat;
+    }
+
+    public void setIat(long iat) {
+        this.iat = iat;
+    }
+
+    public Map<String, EventDetail> getEvents() {
+        if (events == null) {
+            return null;
+        }
+        return new HashMap<>(events);
+    }
+
+    public void setEvents(Map<String, EventDetail> events) {
+        if (events == null) {
+            this.events = null;
+        } else {
+            this.events = new HashMap<>(events);
+        }
+    }
+
+    /**
+     * Represents the detailed information contained within a specific event type.
+     */
+    public static class EventDetail {
+
+        @SerializedName("initiatorType")
+        private String initiatorType;
+
+        @SerializedName("tenant")
+        private Tenant tenant;
+
+        @SerializedName("action")
+        private String action;
+
+
+        public String getInitiatorType() {
+            return initiatorType;
+        }
+
+        public void setInitiatorType(String initiatorType) {
+            this.initiatorType = initiatorType;
+        }
+
+        public Tenant getTenant() {
+            return (this.tenant != null) ? new Tenant(this.tenant) : null;
+        }
+
+        public void setTenant(Tenant tenant) {
+            this.tenant = (tenant != null) ? new Tenant(tenant) : null;
+        }
+
+        public String getAction() {
+            return action;
+        }
+
+        public void setAction(String action) {
+            this.action = action;
+        }
+    }
+
+    /**
+     * Represents the tenant object. This class includes all possible fields from all
+     * event types. Fields not present in a specific JSON will be null.
+     */
+    public static class Tenant {
+
+        @SerializedName("id")
+        private String id;
+
+        @SerializedName("domain")
+        private String domain;
+
+        @SerializedName("owners")
+        private List<Owner> owners;
+
+        @SerializedName("lifecycleStatus")
+        private LifecycleStatus lifecycleStatus;
+
+        @SerializedName("ref")
+        private String ref;
+
+        public Tenant() {
+            
+        }
+
+        public Tenant(Tenant other) {
+            this.id = other.id;
+            this.domain = other.domain;
+            this.ref = other.ref;
+            this.setOwners(other.owners);
+            this.setLifecycleStatus(other.lifecycleStatus); 
+        }
+        
+        public String getId() {
+            return id;
+        }
+
+        public void setId(String id) {
+            this.id = id;
+        }
+
+        public String getDomain() {
+            return domain;
+        }
+
+        public void setDomain(String name) {
+            this.domain = name;
+        }
+
+        public List<Owner> getOwners() {
+            if (this.owners == null) {
+                return null;
+            }
+            return new ArrayList<>(this.owners);
+        }
+
+        public void setOwners(List<Owner> owners) {
```

Comment:
Formatting issue of not leaving a new line after method signature.
Check other places as well.

This formatting issues can be fixed by configuring wso2 code style https://github.com/wso2/product-is/wiki/Contributing-to-Identity-Server#how-to-configure-wso2-code-style-template and formatting the code

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/7075/commits/a6be893041c5d4a898b3d2f74d8525409740cf5c